### PR TITLE
feat: replace team list view with org list view UI and fix team view user availability

### LINKED ISSUE #14646: [CAL-3493] replace team list view with org list view UI
replace

image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3493](https://linear.app/calcom/issue/CAL-3493/replace-team-list-view-with-org-list-view-ui)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #14646 

Screenshot 2024-06-06 20512

Screenshot 2024-09-06 095510

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Good start! It seems like we've removed a few things and not added them back - can you take a look at the invite member modal and the add "CTA" that has been removed. We will still need these

**Comment by anikdhabal** [comment]
> Good start! It seems like we've removed a few things and not added them back - can you take a look at the invite member modal and the add "CTA" that has been removed. We will still need these

The above picture is cropped, and it includes the invite member model and CTA. We handle it in the table.

**Comment by sean-brydon** [comment]
Hey bud thanks for this - i have made some quick UI fixes and pushed them. I have noticed the edit button doesnt work on users so i feel like something has gone wrong here - Could you take a look at this? If not one of us will get to it in the coming days- thanks. 


CleanShot 2024-06-18 at 13 19 48

**Comment by anikdhabal** [comment]
> Hey bud thanks for this - i have made some quick UI fixes and pushed them. I have noticed the edit button doesnt work on users so i feel like something has gone wrong here - Could you take a look at this? If not one of us will get to it in the coming days- thanks.
> 

Thanks for the review. I've fixed that.

**Review by Amit91848 — DISMISSED** [review]
Great work so far @anikdhabal ! Found a few bugs 

- Let's make multiple apps go in a line and add gap between them

  Screenshot 2024-07-05 212043


- Searching the wrong name makes the table completely disappear along with search bar.

   https://github.com/calcom/cal.com/assets/74371312/d6bf71c2-69fc-4a88-8de3-845e885b6c64

**Review by sean-brydon — DISMISSED** [review]
Nice this works well. 

Not blocking but heres a few things i noticed 
CleanShot 2024-08-19 at 10 00 38
this PR is old now so this is deffo not needed to be in this PR but i wonder if we can match the new designs for this in a follow up [figma](https://www.figma.com/design/UajFu4M1APhn2AywAEJkJr/New-Features?node-id=22449-17388&t=MdCAPxBlMEtUhYht-4) This is also in main for orgs/members right now. 

CleanShot 2024-08-19 at 10 00 48
Jumping to this modal from clicking edit is a bit of a jaring experience. I wonder if we can add a transition or move it to the edit sheet like org members 

LGTM - works perfectlly. Feel free to merge and tackle the rest in a follow up if thats alright with you! If not please create an issue to do this in the future

**Comment by anikdhabal** [comment]
Thanks @sean-brydon for the review, Currently, the team availability dialog is also broken and the component is commented out. I will make a follow-up PR and handle these issues.

File: packages/features/ee/teams/components/MemberListItem.tsx [hunk 1]
```diff
@@ -353,26 +602,124 @@ export default function MemberListItem(props: Props) {
...
+            })
+          }
+        />
+      )}
+      {state.inviteMember.showModal && (
+        <MemberInvitationModal
+          isPending={inviteMemberMutation.isPending}
+          isOpen={true}
+          orgMembers={props.orgMembersNotInThisTeam}
+          members={props.members}
```
Comment (Udit-takkar) [inline packages/features/ee/teams/components/MemberListItem.tsx:624]: fetching all the members is not scalable

Comment (Udit-takkar) [reply packages/features/ee/teams/components/MemberListItem.tsx:624]: Can you raise a follow up PR after this is merged https://github.com/calcom/cal.com/pull/16155 and use `MemberInvitationModalWithoutMembers`?

Comment (anikdhabal) [reply packages/features/ee/teams/components/MemberListItem.tsx:624]: Let's wait until your PR is merged. I have one more that depends on this PR being merged. In the meantime, I will fix the other things.

**Comment by anikdhabal** [comment]
After changes
Screenshot 2024-09-02 101416

File: packages/features/ee/teams/components/MemberListItem.tsx [hunk 2]
```diff
@@ -142,242 +262,420 @@ export default function MemberListItem(props: Props) {
...
-                      target="_blank"
-                      href={`${bookerUrl}/${props.member.username}`}
-                      className="text-default block text-sm">
-                      {bookingLink}
-                    </a>
-                  </>
-                )}
+  const memorisedColumns = useMemo(() => {
+    const cols: ColumnDef<User>[] = [
+      // Disabling select for this PR: Will work on actions etc in a follow up
```
Comment (Amit91848) [inline packages/features/ee/teams/components/MemberListItem.tsx:280]: As far as I can see this has been implemented right? Let's remove this comment then
```suggestion
```

**Review by Amit91848 — DISMISSED** [review]
Really solid work @anikdhabal . Found some minor NIT

 - Update button breaks for smaller screens.
   Screenshot 2024-09-06 224241

   Also wouldn't it be better if the cal links and email links were actually clickable? Currently it just normal text 



- https://github.com/user-attachments/assets/ebef4a7f-561c-46b9-93f0-09a02893bba9

   There is some weird scrolling for teams within org. When you go to members tab for teams, it automatically scrolls to the bottom. It is happening consistently with me (Make  sure to have enough members that there is infinite scrolling). Cannot see it happing for org members table.

  Also  if you scroll back to top, change windows and then come back again, it scrolls to the bottom again.

- Org banner covers most of the edit sheet, I guess it could be because the photo was not proper size for banner, but either way we should have a max height for the banner.
   
   Screenshot 2024-09-06 231942

**Comment by anikdhabal** [comment]
@Amit91848 It's okay not to make the car and email a link in the editor. We already have access to visit the member link from the list. The same goes for the banner—since this is a team profile, we don't need to add an organization banner. Just need to make the banner separate. Apart from these, I will fix the other issues.

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
Can't change role of user and button is always loading


https://github.com/user-attachments/assets/2b1a0e98-3d16-4fa1-b8c2-191168f8dfb6

File: packages/features/ee/teams/components/MemberListItem.tsx [hunk 3]
```diff
@@ -142,242 +262,420 @@ export default function MemberListItem(props: Props) {
...
-                    variant="icon"
-                    StartIcon="external-link"
-                    disabled={!props.member.accepted}
-                  />
-                </Tooltip>
+          );
+        },
+        filterFn: (rows, id, filterValue) => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore Weird typing issue
```
Comment (Udit-takkar) [inline packages/features/ee/teams/components/MemberListItem.tsx:335]: can we fix this?

File: packages/features/ee/teams/components/MemberListItem.tsx [hunk 4]
```diff
@@ -410,28 +716,76 @@ export default function MemberListItem(props: Props) {
...
+                showModal: true,
+              },
+            });
+            dispatch({
+              type: "INVITE_LINK_SETTING",
+              payload: {
+                showModal: false,
+              },
+            });
+          }}
```
Comment (Udit-takkar) [inline packages/features/ee/teams/components/MemberListItem.tsx:764]: why did you move this modal inside `MemberListItem`  ? we should keep the modal outside of `MemberListItem` as it doesn't depend upon any member

File: packages/features/ee/teams/components/MemberListItem.tsx [hunk 5]
```diff
@@ -23,93 +29,208 @@ import {
...
+  const [searchTerm, setSearchTerm] = useState("");
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+  const [connectedApps, setConnectedApps] = useState<Record<number, ConnectedAppsType[]>>({});
+  const [userIds, setUserIds] = useState<number[]>([]);
+
+  const { data: getUserConnectedApps } = trpc.viewer.teams.getUserConnectedApps.useQuery(
+    { userIds, teamId: props.team.id },
+    { enabled: !!userIds.length }
+  );
+
```
Comment (Udit-takkar) [inline packages/features/ee/teams/components/MemberListItem.tsx:152]: if we are displaying connected apps only in edit modal then we should not fetch it before the modal is clicked.

Comment (anikdhabal) [reply packages/features/ee/teams/components/MemberListItem.tsx]: Yeah, make sense

File: packages/features/ee/teams/components/MemberListItem.tsx (hunk 4 above)
Comment (anikdhabal) [reply packages/features/ee/teams/components/MemberListItem.tsx]: Also, the memberinvitationmodel

**Comment by anikdhabal** [comment]
> Can't change role of user and button is always loading
> 

This is different issue we have currently. Fixed it

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**